pilot/pkg/model/sidecar.go

```diff
@@ -355,6 +356,8 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
 			// Update index as well, so that future reads will merge into the new service
 			foundSvc.svc = copied
 			servicesAdded[foundSvc.svc.Hostname] = foundSvc
+			// update the existing service in the map to the merged one
+			out.servicesByHostname[s.Hostname] = copied
```

Comment:
Sorry, if we remove this, `TestCreateSidecarScope` (in sidecar_test.go) seems still successful. Do we miss something?

Reply:
part of https://github.com/istio/istio/pull/50691

Reply:
I acknowledge that, but it seems the test in this version doesn't validate the claimed assertion here: https://github.com/istio/istio/pull/50691#discussion_r1581995972.

Reply:
make sense, reverted